src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -865,6 +870,11 @@ await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '.')
                     c = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
                 }
 
+                if (response.Headers.ConnectionClose ?? false)
```

Comment:
Have you verified the perf impact of this?  This is likely going to allocate the general headers in the response if it hasn't been allocated already:
https://github.com/dotnet/corefx/blob/6bf1e7c3032d3c24d5a41bbd880549ef4a8de2da/src/System.Net.Http/src/System/Net/Http/Headers/HttpResponseHeaders.cs#L100
I made the general headers object lazily allocated such that we could avoid things like this in the handler code and it would only be allocated if user code accessed it.  If we want to do this check, it'd be good to implement a workaround for the allocation like I did here:
https://github.com/dotnet/corefx/blob/6bf1e7c3032d3c24d5a41bbd880549ef4a8de2da/src/System.Net.Http/src/System/Net/Http/Headers/HttpResponseHeaders.cs#L127
https://github.com/dotnet/corefx/commit/79ae7ebf5f7e792e35c9b68af6d148b5ef24c818
Alternatively, instead of looking up the header afterwards, you could measure the impact of just tracking this as we read the headers, since we're the ones populating the response headers to begin with.